### DISCUSSION TITLE
How can I obtain the information about the exception that the current Agent is verified by the verifier?

### DISCUSSION THREAD

**Discussion by webfast-lab** [keylime/keylime#1691] [General]
Currently, I only see the log information, but no information is found in the database.If I manage many nodes, I can only query the current node status. If a node is faulty, I can only query the log information.

**Comment by THS-on** [comment]
You can get the status of an agent with:
```
keylime_tenant -c cvstatus
```
This gives you this output (I formatted the JSON a bit):
```
{
  "d432fbb3-d2f1-4a97-9ef7-75bd81c00000": {
    "operational_state": "Get Quote",
    "v": null,
    "ip": "127.0.0.1",
    "port": 9002,
    "tpm_policy": "{\"mask\": \"0x0\"}",
    "meta_data": "{}",
    "has_mb_refstate": 0,
    "has_runtime_policy": 0,
    "accept_tpm_hash_algs": [
      "sha512",
      "sha384",
      "sha256",
      "sha1"
    ],
    "accept_tpm_encryption_algs": [
      "ecc",
      "rsa"
    ],
    "accept_tpm_signing_algs": [
      "ecschnorr",
      "rsassa"
    ],
    "hash_alg": "sha256",
    "enc_alg": "rsa",
    "sign_alg": "rsassa",
    "verifier_id": "default",
    "verifier_ip": "127.0.0.1",
    "verifier_port": 8881,
    "severity_level": null,
    "last_event_id": null,
    "attestation_count": 16,
    "last_received_quote": 1732637518,
    "last_successful_attestation": 1732637518
  }
}
```
What relevant for you here is:
* `last_successful_attestation`: when the last successfully attestation was
* `attestation_count`: number of successful attestation rounds
* `last_event_id`: If there was a failure, which type is was (see also https://github.com/keylime/enhancements/blob/master/46_revocation_severity_and_context.md)
* `severity_level`: Severity of the failure

If you want to get this info for all the agents registered with a verifier run:
```
keylime_tenant -c bulkinfo
```

In the case of a failure you can notified via a webhook mechanism: https://github.com/keylime/keylime/blob/f1514b6f4a882dd7ed73301d028212f9dcc9bed1/templates/2.4/verifier.j2#L229-L252
